### ISSUE TITLE
Update to Serde 1.0

### ISSUE THREAD

**Issue by dtolnay** [OneSignal/serde-redis#10]
I know you just did the 0.9 upgrade but hopefully this is the last one for a long time. Release notes:
https://github.com/serde-rs/serde/releases/tag/v1.0.0

**Comment by jwilm** [comment]
Haha, yep!

I actually have a question about the zero-copy deserialization that I haven't looked into yet. Is it possible to opt out? This deserializer actually operates one level abstracted from the protocol where duplicate values have already been made. To be clear, there's a `Value` enum like this:

```rust
enum Value {
    Bulk(Vec<Value>),
    Data(Vec<u8>),
    // ...
}
```

which already includes allocations. Switching to zero copy would prevent the deserializer impl moving out of the `Value` enum would and require an additional allocation for owned types.

I'd like to change this crate to operate at the protocol level eventually (at which point zero-copy makes sense), but for now it is easier to stick with moving data out of the `Value` enum.

**Comment by dtolnay** [comment]
Yes, if you change the signature of `serde_redis::from_redis_value` to require [`T: DeserializeOwned`](https://docs.serde.rs/serde/de/trait.DeserializeOwned.html) then it disallows any Deserialize impls that would want to borrow from the input. This is what [`from_reader`](https://docs.serde.rs/serde_json/fn.from_reader.html) uses in serde_json, for example.

Aside from that, your Deserializer can call Visitor::visit_string just like before and transfer ownership of string data.

**Comment by jwilm** [comment]
> Yes, if you change the signature of `serde_redis::from_redis_value` to require [`T: DeserializeOwned`](https://docs.serde.rs/serde/de/trait.DeserializeOwned.html) then it disallows any Deserialize impls that would want to borrow from the input. This is what [`from_reader`](https://docs.serde.rs/serde_json/fn.from_reader.html) uses in serde_json, for example.

Ah, great tip. Thanks!

> Aside from that, your Deserializer can call Visitor::visit_string just like before and transfer ownership of string data.

Perfect.

**Comment by dtolnay** [comment]
Fixed in #11.
